plenoptic/simulate/models/frontend.py

```diff
@@ -1,96 +1,421 @@
+from typing import Tuple, Union, Callable
+
 import torch
 import torch.nn as nn
-from torchvision import transforms
-from ...tools.signal import make_disk
-import os
+import torch.nn.functional as F
+from torch import Tensor
+
 from ...tools.display import imshow
+from ...tools.signal import make_disk
+
+from collections import OrderedDict
+
+__all__ = ["Gaussian", "CenterSurround", "LN", "LG", "LGG", "OnOff"]
+
 
+def get_pad(kernel_size: Union[int, Tuple[int, int]]) -> Tuple[int, int, int, int]:
```

Comment:
Are you sure this will work for both even and odd-sized kernels? I guess this again depends on what padding we want as default, but depending on if it's even or odd-size I think that would require different paddings for example if you wanted to keep the same output size.

Reply:
Added new padding method in `tools/conv.py` that u helped me with